api/include/opentelemetry/trace/span.h

```diff
@@ -10,7 +10,13 @@
 #include "opentelemetry/trace/key_value_iterable_view.h"
 #include "opentelemetry/version.h"
 
+#define SPAN_KEY "span_key"
```

Comment:
nit: we don't have a very strict rule on macro usage, but ideally this should be `constexpr cost char *` in opentelemetry namespace instead of a macro. I think we mentioned briefly we want to more or less follow C++ Core Guidelines : https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es31-dont-use-macros-for-constants-or-functions ..

Guidelines: `Scream when you see a macro that isn’t just used for source control (e.g., #ifdef)`
Me: 😱

Reply:
Changed to const char[].